src/main/java/uk/ac/sanger/sccp/stan/service/history/HistoryServiceImp.java

```diff
@@ -1062,6 +1070,16 @@ public List<HistoryEntry> createEntriesForDestructions(Collection<Destruction> d
         return entries;
     }
 
+    /**
+     * Gets a list of the barcodes of the indicated labware that are flagged
+     * @param labware the labware
+     * @return the barcodes of those labware that are flagged
+     */
+    public List<String> loadFlaggedBarcodes(Collection<Labware> labware) {
+        List<LabwareFlagged> lfs = flagLookupService.getLabwareFlagged(labware);
+        return lfs.stream().filter(LabwareFlagged::isFlagged).map(lf -> lf.getLabware().getBarcode()).toList();
+    }
+
     /**
```

Comment:
Would it be more optimal to return the list of flagged labware directly instead of returning both the list of labware and the barcode list of the flagged labware?

Reply:
The method returns the list of barcodes because that’s what the graphql response requires. We could have the graphql response include LabwareFlagged instead of Labware, but that would need more substantial changes in the client, and would only give you a smaller response if most of the labware are flagged.